mars/dataframe/reduction/core.py

```diff
@@ -0,0 +1,246 @@
+    @classmethod
+    def _get_reduced_meta(cls, chunk, axis):
+        """
+        calculate meta information of reduced result.
+        :param chunk: chunk will be reduced
+        :param axis: axis
+        :return:
+        For series, return (shape, index, dtype).
+        For DataFrame, return (shape, index, dtypes).
+        """
+        if isinstance(chunk, SERIES_CHUNK_TYPE):
+            return (), parse_index(pd.RangeIndex(0)), chunk.dtype
+        if axis == 0 or axis == 'index':
+            return (1, chunk.shape[1]), parse_index(pd.RangeIndex(1)), chunk.dtypes
+        else:
+            emtpy_df = build_empty_df(chunk.dtypes)
+            func_name = getattr(cls, '_func_name')
+            reduced_dtypes = pd.Series(getattr(emtpy_df, func_name)(axis=axis).dtypes)
+            return (chunk.shape[0], 1), chunk.index_value, reduced_dtypes
+
+    @classmethod
+    def _tile_one_chunk(cls, op):
+        df = op.outputs[0]
+        chk = op.inputs[0].chunks[0]
+        new_chunk_op = op.copy().reset_key()
+        reduced_shape, index_value, dtype = cls._get_reduced_meta(chk, axis=op.axis)
+        chunk = new_chunk_op.new_chunk(op.inputs[0].chunks, shape=reduced_shape, index=chk.index,
+                                       index_value=index_value, dtype=dtype)
+        new_op = op.copy()
+        nsplits = tuple((s,) for s in chunk.shape)
+        return new_op.new_seriess(op.inputs, df.shape, nsplits=nsplits, chunks=[chunk],
+                                  index_value=df.index_value, dtype=df.dtype)
+
+    @classmethod
+    def _build_reduction_chunks(cls, op):
+        chunks = np.empty(op.inputs[0].chunk_shape, dtype=np.object)
+        for c in op.inputs[0].chunks:
+            new_chunk_op = op.copy().reset_key()
+            if isinstance(c, SERIES_CHUNK_TYPE):
+                reduced_shape, index_value, dtype = cls._get_reduced_meta(c, axis=op.axis)
+                chunks[c.index] = new_chunk_op.new_chunk([c], shape=reduced_shape, index=c.index,
+                                                         dtype=dtype, index_value=index_value)
+            else:
+                new_chunk_op._object_type = ObjectType.dataframe
+                reduced_shape, index_value, dtypes = cls._get_reduced_meta(c, axis=op.axis)
+                chunks[c.index] = new_chunk_op.new_chunk([c], shape=reduced_shape, index=c.index,
+                                                         dtypes=dtypes, index_value=index_value)
+        return chunks
+
+
+class DataFrameReductionMixin(ReductionMixin):
+    @classmethod
+    def tile(cls, op):
+        df = op.outputs[0]
+        in_df = op.inputs[0]
+        combine_size = op.combine_size or options.combine_size
+
+        if len(in_df.chunks) == 1:
+            return cls._tile_one_chunk(op)
+
+        n_rows, n_cols = in_df.chunk_shape
+        reduction_chunks = cls._build_reduction_chunks(op)
+        out_chunks = []
+        if op.axis is None or op.axis == 0 or op.axis == 'index':
```

Comment:
`op.axis` should be written as 0 if it's `index`, 1 if it's `columns` when calling `__call__` to reduce the code.